xapian-core/api/leafpostlist.cc

```diff
@@ -72,14 +73,16 @@ double
 LeafPostList::get_weight() const
 {
     if (!weight) return 0;
-    Xapian::termcount doclen = 0, unique_terms = 0;
+    Xapian::termcount doclen = 0, unique_terms = 0, wdfdocmax = 0;
     // Fetching the document length and number of unique terms is work we can
     // avoid if the weighting scheme doesn't use them.
     if (need_doclength)
 	doclen = get_doclength();
     if (need_unique_terms)
 	unique_terms = get_unique_terms();
-    double sumpart = weight->get_sumpart(get_wdf(), doclen, unique_terms);
+    if (need_wdfdocmax)
+    wdfdocmax = get_wdfdocmax();
```

Comment:
Here's an example of mis-identation - the line after the `if` should be indented more.